### PR TITLE
Services for add edit park.

### PR DESCRIPTION
**Pull request by danieltruong**

### CODE REVIEW

**Review by marklise — APPROVED** [review]

File: src/app/park-form/park-form.component.html [hunk 1]
```diff
@@ -1,111 +1,149 @@
...
-                        <div class="input-group">
-                            <input id="link" formControlName="link" class="form-control" type="text"
-                                placeholder="Paste link here">
-                            <div class="input-group-append">
-                                <button (click)="testLink()" [disabled]="!parkForm.get('link').value" class="btn btn-primary" type="button">Test Link</button>
-                            </div>
+                </div>
+            </div>
+
+            <div class="row">
```
Comment (cameronpettit) [inline src/app/park-form/park-form.component.html:74]: disable the park name field when editing park - changing name creates new park

File: src/app/park-form/park-form.component.html [hunk 2]
```diff
@@ -1,111 +1,149 @@
...
+                    <button class='btn btn-danger mr-1'
+                        (click)="cancel()">Cancel</button>
+                    <button class='btn btn-secondary mx-1'
+                        [disabled]="parkForm.pristine"
+                        (click)="resetForm()">Reset
                         Form</button>
-                    <button class='btn btn-primary m-1' [disabled]="parkForm.invalid" (click)="submitForm()">Save
+                    <button class='btn btn-primary ml-1'
+                        [disabled]="parkForm.invalid"
+                        (click)="submitForm()">Save
```
Comment (cameronpettit) [inline src/app/park-form/park-form.component.html:143]: this row of buttons is not properly right justified, but only on park-add (park-edit is fine) - i think removing the delete button in add is causing this

**Review by cameronpettit — APPROVED** [review]
Minor changes that can be dealt with in polish

File: src/app/park-form/park-form.component.html (hunk 2 above)
Comment (danieltruong) [reply src/app/park-form/park-form.component.html:143]: 068e8a9f780f2d0e3f81771cb8ee53f01755fa68

File: src/app/park-form/park-form.component.html (hunk 1 above)
Comment (danieltruong) [reply src/app/park-form/park-form.component.html:74]: 068e8a9f780f2d0e3f81771cb8ee53f01755fa68

**Pull request merged**